go/interpreter/lfvm/instructions.go

```diff
@@ -1084,19 +1091,20 @@ func opCall(c *context) {
 	// and not doing it would be identified by the replay tool as an error.
 	toAddr := tosca.Address(addr.Bytes20())
 
+	// for static and delegate calls, the following value checks will always be zero.
+
 	// Charge for transferring value to a new address
 	if !value.IsZero() {
 		base_gas += tosca.Gas(params.CallValueTransferGas)
 	}
 
 	// if evm.chainRules.IsEIP158 according to GETH it is EIP158 since 2016
 	// !!!! but need to touch stateDB for the address to have it in the substate record key/value
-	if !value.IsZero() && !c.context.AccountExists(toAddr) {
+	if !value.IsZero() && !c.context.AccountExists(toAddr) && kind == tosca.Call {
 		base_gas += tosca.Gas(params.CallNewAccountGas)
 	}
 
 	cost := callGas(c.gas, base_gas, provided_gas)
-
 	if warmAccess {
```

Comment:
This whole if could be refactor: 

```
    if !warmAccess {
                // In case of a cold access, we temporarily add the cold charge back, and also
		// add it to the returned gas. By adding it to the return, it will be charged
		// outside of this function, as part of the dynamic gas, and that will make it
		// also become correctly reported to tracers.
		c.gas += coldCost
                base_gas += coldCost
     }

     if !c.UseGas(base_gas + cost) {
	return
     }
```

- base_gas is never used afterwards
- could you rename base_gas to baseGas?